applications/PfemFluidDynamicsApplication/python_scripts/KratosPfemFluid.py

```diff
@@ -1,234 +1,355 @@
 
+        print(" ")
 
-#import define_output
-parameter_file = open("ProjectParameters.json",'r')
-PProjectParameters =  Parameters(parameter_file.read())
+        # defining the number of threads:
+        num_threads = self.ProjectParameters["problem_data"]["threads"].GetInt()
+        self.SetParallelSize(num_threads)
+        print("::[KPFEM Simulation]:: [OMP USING",num_threads,"THREADS ]")
+        #parallel.PrintOMPInfo()
 
-##set echo level
-echo_level = PProjectParameters["problem_data"]["echo_level"].GetInt()
 
-print(" ")
+        print(" ")
+        print("::[KPFEM Simulation]:: [Time Step:", self.ProjectParameters["problem_data"]["time_step"].GetDouble()," echo:", self.echo_level,"]")
 
-# defining the number of threads:
-threads = PProjectParameters["problem_data"]["threads"].GetInt()
-SetParallelSize(threads)
+        #### Model_part settings start ####
 
-print(" ")
-print("::[KPFEM Simulation]:: [Time Step:", PProjectParameters["problem_data"]["time_step"].GetDouble()," echo:", echo_level,"]")
+        # Defining the model_part
+        self.main_model_part = KratosMultiphysics.ModelPart(self.ProjectParameters["problem_data"]["model_part_name"].GetString())
 
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE, self.ProjectParameters["problem_data"]["domain_size"].GetInt())
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DELTA_TIME, self.ProjectParameters["problem_data"]["time_step"].GetDouble())
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.TIME, self.ProjectParameters["problem_data"]["start_time"].GetDouble())
+        if( self.ProjectParameters["problem_data"].Has("gravity_vector") ):
+            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.GRAVITY_X, self.ProjectParameters["problem_data"]["gravity_vector"][0].GetDouble())
+            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.GRAVITY_Y, self.ProjectParameters["problem_data"]["gravity_vector"][1].GetDouble())
+            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.GRAVITY_Z, self.ProjectParameters["problem_data"]["gravity_vector"][2].GetDouble())
+
+        ###TODO replace this "model" for real one once available in kratos core
+        self.Model = {self.ProjectParameters["problem_data"]["model_part_name"].GetString() : self.main_model_part}
+
+        #construct the solver (main setting methods are located in the solver_module)
+        solver_module = __import__(self.ProjectParameters["solver_settings"]["solver_type"].GetString())
+        self.solver = solver_module.CreateSolver(self.main_model_part, self.ProjectParameters["solver_settings"])
+
+
+        #### Output settings start ####
+
+        self.problem_path = os.getcwd()
+        self.problem_name = self.ProjectParameters["problem_data"]["problem_name"].GetString()
+        
+        self.SetGraphicalOutput()   
```

Comment:
It raises an errror, self.computing_model_part is needed and it is created in the Initilialize...the position of this call must be after it.